src/compiler/checker.ts

```diff
@@ -4859,12 +4884,21 @@ namespace ts {
 
                 // Note that the "each" checks must precede the "some" checks to produce the correct results
                 if (source.flags & TypeFlags.Union) {
-                    if (result = eachTypeRelatedToType(<UnionType>source, target, reportErrors)) {
+                    if (relation === comparableRelation) {
+                        result = someTypeRelatedToType(source as UnionType, target, reportErrors);

```

Comment:
Please choose one cast syntax and make at least the changed lines use it, preferably something like all the casts in the function.